Mathlib/Probability/Distributions/Gaussian.lean

```diff
@@ -347,6 +348,52 @@ lemma gaussianReal_mul_const {X : Ω → ℝ} (hX : Measure.map X ℙ = gaussian
 
 end Transformations
 
+open Measurable Real
+
+variable {Ω : Type*} {mΩ : MeasurableSpace Ω} {p : Measure Ω} {μ : ℝ} {v : ℝ≥0} {X : Ω → ℝ}
+
+lemma mgf_dirac (hX : p.map X = .dirac μ) (t : ℝ) : mgf X p t = exp (μ * t) := by
+  rw [← mgf_id_map, mgf, hX, integral_dirac, mul_comm]
+  · rfl
+  · apply AEMeasurable.of_map_ne_zero
+    rw [hX]
+    exact IsProbabilityMeasure.ne_zero (.dirac μ)
+
+theorem mgf_gaussianReal (hX : p.map X = gaussianReal μ v) (t : ℝ) :
+    mgf X p t = exp (μ * t + v * t ^ 2 / 2) := by
+  by_cases hv : v = 0
+  · simp only [gaussianReal, hv, ↓reduceIte] at hX
+    rw [mgf_dirac hX, hv]
+    simp only [NNReal.coe_zero, zero_mul, zero_div, add_zero]
+  · calc
+    mgf X p t = (p.map X)[fun x => exp (t * x)] := by
+      rw [← mgf_id_map, mgf]
+      · rfl
+      · apply AEMeasurable.of_map_ne_zero
+        rw [hX]
+        exact IsProbabilityMeasure.ne_zero (gaussianReal μ v)
+    _ = ∫ x, exp (t * x) * gaussianPDFReal μ v x := by
+      rw [hX, gaussianReal_of_var_ne_zero μ hv, gaussianPDF_def]
+      simp only [ENNReal.ofReal]
+      rw [integral_withDensity_eq_integral_smul
+        (measurable_gaussianPDFReal μ v).real_toNNReal]
+      congr with x
+      rw [NNReal.smul_def, coe_toNNReal _ (gaussianPDFReal_nonneg μ v x), smul_eq_mul, mul_comm]
```

Comment:
You should not be shy to chuck a bunch of stuff into a simp call
```suggestion
    simp [integral_withDensity_eq_integral_smul (measurable_gaussianPDFReal μ v).real_toNNReal, hX,
      gaussianReal_of_var_ne_zero μ hv, gaussianPDF_def, ENNReal.ofReal, NNReal.smul_def, mul_comm]
```

Reply:
Huh, I thought I tried that and it did not work. Thanks!